app/Http/Controllers/HomeController.php

```diff
@@ -68,25 +69,51 @@ public function getDownload()
             'windows_x64' => osu_trans('home.download.os_version_or_later', ['os_version' => 'Windows 8.1']).' (x64)',
         ];
 
-        $deviceDetector = new DeviceDetector(\Request::header('User-Agent') ?? '');
-        $deviceDetector->parse();
-        $family = $deviceDetector->getOs('family');
-
-        $platform = match ($family) {
-            // Try matching most likely platform first
-            'Windows' => 'windows_x64',
-            // current iPadOS declares itself as a desktop browser.
-            'iOS' => 'ios',
-            // FIXME: Figure out a way to differentiate Intel and Apple Silicon.
-            'Mac' => 'macos_as',
-            'Android' => 'android',
-            'GNU/Linux' => 'linux_x64',
-            default => 'windows_x64',
-        };
+        $platform = request('platform');
+        if (!array_key_exists($platform, $lazerPlatformNames)) {
+            $platform = null;
+        }
+
+        if ($platform === null) {
```

Comment:
might as well just do it in one step?

```suggestion
        if (!array_key_exists($platform, $lazerPlatformNames)) {
```